cmd/osm-controller/osm-controller.go

```diff
@@ -240,8 +240,7 @@ func main() {
 		msgBroker,
 	)
 
-	proxyMapper := &registry.KubeProxyServiceMapper{KubeController: k8sClient}
-	proxyRegistry := registry.NewProxyRegistry(proxyMapper, msgBroker)
+	proxyRegistry := registry.NewProxyRegistry()
```

Comment:
So it seems like the gist of this change is that the proxyRegistry loses a dependency on a `proxyMapper` and instead relies on the new `GetServicesForProxy` method declared on the client (accessible to CDS via the meshcatalog's k8s client instead of the proxy registry). Is that correct? If so, it seems like this changeset further decouples our Envoy config from k8s primitives, increasing the use of our domain objects. Nice!

Reply:
that's correct!